src/native/libs/System.IO.Compression.Native/pal_zlib.c

```diff
@@ -44,6 +39,10 @@ Initializes the PAL_ZStream by creating and setting its underlying z_stream.
 static int32_t Init(PAL_ZStream* stream)
 {
     z_stream* zStream = (z_stream*)calloc(1, sizeof(z_stream));
+
+	zStream->zalloc = z_custom_calloc;
+	zStream->zfree = z_custom_cfree;
```

Comment:
Where was custom allocator integration being done previously?

Reply:
Here: https://github.com/dotnet/runtime/pull/84604

Reply:
We had to rename these methods because we were seeing symbol conflict errors between zlib-ng and zlib. Instead of being resolved by symbol name, we explicitly provided them to the zlib engine.

Reply:
> We had to rename these methods because we were seeing symbol conflict errors between zlib-ng and zlib

Was this the only symbol conflict? If you get both zlib-ng and zlib on one linker command line, I expect that you will get a whole bunch of symbol conflicts. These two would be only the tip of the iceberg.

Reply:
IIRC I was also seeing the other zlib public methods conflicting (deflateInit, crc32, etc). This happened very early while working on this PR.